### DISCUSSION TITLE
500 error on sending automatic mail to users on creating a new item (like a blog)

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#2762] [Archive]
I'm trying to understand how I the Hooks in Directus work and what I have to do to mail all registered users when I create a new item (it's a for a personal blog). I've read the documentation on your website and tried the code below:
`<?php return [ 'actions' => [ 'item.create.articles' => function ($data) { $content = 'New article was created with the title: ' . $data['title']; notify('myname@gmail.com', 'New Article', $content); }]]`
but it doesn't work. I get a 500 error.

Someone can please help me?

**Comment by rijkvanzanten** [comment]
That "notify" function is just an example of where you can inject your own logic. You could use something like `mail` in PHP (https://www.php.net/manual/en/function.mail.php) or rely on an external service like [Sendgrid](https://sendgrid.com/) etc.

**Comment by ghost** [comment]
Aah okay. You maybe have an example of  mail in PHP?

Op ma 22 jun. 2020 om 17:32 schreef Rijk van Zanten <
notifications@github.com>:

**Reply by rijkvanzanten** [reply]
See my response above, I already provided a couple 🙂 

> You could use something like mail in PHP (https://www.php.net/manual/en/function.mail.php) or rely on an external service like Sendgrid etc.

**Comment by ghost** [comment]
Yeah okay. But how to use it with directus was more my question 😅

Op di 23 jun. 2020 18:10 schreef Rijk van Zanten <notifications@github.com>:

**Comment by ghost** [comment]
[image: image.png]

This code above isn't working

Op di 23 jun. 2020 om 18:10 schreef Rijk van Zanten <
notifications@github.com>:
